src/Npgsql/NpgsqlConnector.cs

```diff
@@ -210,12 +210,13 @@ internal void FlagAsWritableForMultiplexing()
 
         bool _sendResetOnClose;
 
+        ConnectorPool? _pool;
+
         /// <summary>
-        /// If pooled, the pool index on which this connector will be returned to the pool.
+        /// Contains the UTC timestamp when this connector was opened, used to implement
+        /// <see cref="NpgsqlConnectionStringBuilder.ConnectionLifetime"/>.
         /// </summary>
-        internal int PoolIndex { get; set; } = int.MaxValue;
-
-        ConnectorPool? _pool;
+        internal DateTime OpenTimestamp { get; set; }
```

Comment:
Would like to see a private setter, but makes sense to review other props too outside of this PR.

Reply:
Good point - changed this. And yeah, we should review other props, it's a bit messy.